src/components/ErrorBoundary.tsx

```diff
@@ -0,0 +1,51 @@
+import { Component, type ReactNode } from 'react';
+import '../styles/ErrorBoundary.scss';
+
+interface Props {
+  children: ReactNode;
+}
+
+interface State {
+  message: string | null;
+}
+
+class ErrorBoundary extends Component<Props, State> {
+  timeoutId: number | null = null;
+
+  constructor(props: Props) {
+    super(props);
+    this.state = { message: null };
+  }
+
+  componentDidMount() {
+    window.addEventListener('GlobalError', this.handleGlobalError);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('GlobalError', this.handleGlobalError);
+    if (this.timeoutId) clearTimeout(this.timeoutId);
+  }
+
+  handleGlobalError = (e: Event) => {
+    const customEvent = e as CustomEvent;
+    const message = customEvent.detail?.message || 'Unknown error occurred';
+    this.setState({ message });
+
+    this.timeoutId = window.setTimeout(() => {
+      this.setState({ message: null });
+    }, 5000);
+  };
```

Comment:
ErrorBoundary wraps the entire application, it will not be unmounted and removeEventListener will not be executed and `componentWillUnmount` will never be trigger
`componentDidMount` - not needed when intercepting errors!
In general, all this is not necessary.
but needed `componentDidCatch`

```suggestion
   componentDidCatch(error: unknown, info: React.ErrorInfo) {
    const message = error instanceof Error ? error.message : 'Unknown error';
    this.setState({ error, message });
    console.log(error, info);
  }
```